tensorboard/data/provider.py

```diff
@@ -486,6 +516,78 @@ def __repr__(self):
         )
 
 
+class HyperparameterDomainType(enum.Enum):
+    """Describes how to represent the set of known values for a hyperparameter."""
+
+    # A range of numeric values. Normally represented as Tuple[float, float].
+    INTERVAL = "interval"
+    # A finite set of numeric values. Normally represented as List[float].
+    DISCRETE_FLOAT = "discrete_float"
+    # A finite set of string values. Normally represented as List[string].
+    DISCRETE_STRING = "discrete_string"
+    # A finite set of bool values. Normally represented as List[bool].
+    DISCRETE_BOOL = "discrete_bool"
+
+
+class Hyperparameter:
```

Comment:
Should this be a dataclass (`@dataclasses.dataclass(frozen=True)`)?

Reply:
We don't use dataclass elsewhere in this interface. Now that we require Python 3.8+, though, it could be worth upgrading the entire file to use dataclass  -- I've added it as a task in our task list.